src/test/java/com/conveyal/datatools/manager/jobs/ArbitraryTransformJobTest.java

```diff
@@ -0,0 +1,271 @@
+        feedSource.transformRules.add(transformRules);
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+        // Create new version. This will handle processing the initial version, cloning that version, and applying the
+        // transformations to the cloned version. It will conclude once the cloned version is processed.
+        sourceVersion = createFeedVersion(
+            feedSource,
+            zipFolderFiles("fake-agency-with-only-calendar")
+        );
+        // Grab the cloned version and check assertions.
+        targetVersion = feedSource.retrieveLatest();
+        LOG.info("Checking canCloneZipFileAndTransform assertions.");
+        assertEquals(
+            "Cloned version number should increment by one over original version.",
+            sourceVersion.version + 1,
+            targetVersion.version
+        );
+        assertEquals(
+            "Cloned version retrieval method should be VERSION_CLONE",
+            targetVersion.retrievalMethod,
+            VERSION_CLONE
+        );
+        assertEquals(
+            "feed_info.txt row count should equal input csv data # of rows",
+            2, // Magic number should match row count in string produced by generateFeedInfo
+            targetVersion.feedLoadResult.feedInfo.rowCount
+        );
+        // Check for presence of new feedId in database (one record).
+        assertThatSqlCountQueryYieldsExpectedCount(
+            String.format(
+                "SELECT count(*) FROM %s.feed_info WHERE feed_id = '%s'",
+                targetVersion.namespace,
+                feedId
+            ),
+            1
+        );
+    }
+
+    @Test
+    public void replaceGtfsPlusFileFailsIfSourceIsMissing() throws IOException {
+        sourceVersion = createFeedVersion(
+            feedSource,
+            zipFolderFiles("fake-agency-with-only-calendar")
+        );
+        // Replace file transformation runs before feed is loaded into database.
+        // Note: stop_attributes.txt is a GTFS+ file found in BART's feed.
+        FeedTransformation transformation = ReplaceFileFromVersionTransformation.create(sourceVersion.id, "realtime_routes");
+        FeedTransformRules transformRules = new FeedTransformRules(transformation);
+        feedSource.transformRules.add(transformRules);
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+        // Create new BART version (note: bart_new.zip GTFS file has been stripped of stop_attributes.txt)
+        targetVersion = createFeedVersion(
+            feedSource,
+            zipFolderFiles("fake-agency-with-only-calendar-dates")
+        );
+        // TODO Check that new version has stop_attributes file that matches source version's copy.
+        assertThat(targetVersion.validationResult, Matchers.nullValue());
+    }
+
+    @Test
+    public void canReplaceFeedInfo() throws SQLException, IOException {
```

Comment:
On Windows you get the following warning which in turn fails this test:

java.nio.file.FileSystemException: C:\tmp\Test_Feed_Tue_Jun_30_145808_BST_2020-20200630T145809-01-619a856f-4b4d-4abb-a267-8eb353141241-1af5e4ae-c948-4320-8843-85c6beeaf5de.zip: The process cannot access the file because it is being used by another process.

This is for reference. Perhaps if I move on to working on datatools I'll attempt to resolve.